Manual: build fix

Octachron opened the pull request:
This PR makes the manual build again:
- The first commit fixes a problem with an odd number of `"` inside `cmds/intf-c.etex`
- The second commit fixes an assertion failure within ocamldoc. However, I am worried that I have
  no idea why this assertion failure was triggered in the first place.

Octachron committed:
Manual: fix quotes in intf-c.etex

Octachron committed:
Ocamldoc: display inline record within exceptions

This commit adds a very basic handling of inline record within
exceptions for signature items. Previously, a signature like

sig
  exception E of {lbl:int}
end

would crash OCamldoc due to an assertion failure.
With this commit, the previous signature should be correctly
displayed by ocamldoc as

sig
  exception E of {lbl:int}
end

Beware, that the same exception as structure item would still
crash ocamldoc at the time of this commit.

gasche commented:
This feels like code pattern-matching on the arguments of a constructor declaration, assuming that they are parsed in tuple-like form, and failing on new code using inline records.

I am frustrated by the fact that I have no idea if the code you used to replace the `TODO` part is correct. Could you maybe add a testcase where this call is exercised, and in the commit message tell us what used to be printed and what you now get?

Octachron committed:
Ocamldoc: test inline record within exceptions

Beware: hese tests are incomplete, due to the incomplete handling
of inline record within exception inside Ocamldoc.

Octachron committed:
Testsuite, ocamldoc: test display of inline record

This commit adds a new tests focused on the handling of inline records
by the latex generator. For now, it checks that the inline records
labels and types are at least printed within the latex file and not
replaced by `{...}`.

Octachron commented:
With further investigations, I discovered that the manual compilation was failing on the documentation for https://github.com/ocaml/ocaml/blob/trunk/utils/misc.mli#L312, due to the incomplete handling of
inline record within exceptions.

I have added two incomplete tests for these inline records in `tool-ocamldoc/t04.ml` for the general
handling of exception with inline record and `tools-ocamldoc-2/inline_records.mli` for the handling
of inline records by the latex generator.

Note that this pull request is only a partial fix, and there is still some assertion failure waiting to happen for inline records within ocamldoc, see for instance the commented example in `t04.ml`. Another point that might be worth to look at in the future is that for now, inline record discards field documentation comments.

gasche commented:
Excellent, thanks!

I created a MPR for it: [MPR#7290](http://caml.inria.fr/mantis/view.php?id=7290).
